Zend/zend_property_hooks.h

```diff
@@ -23,19 +23,21 @@
 
 BEGIN_EXTERN_C()
 
-typedef struct _zend_array zend_array;
-typedef struct _zend_class_entry zend_class_entry;
-typedef struct _zend_object zend_object;
-typedef struct _zend_object_iterator zend_object_iterator;
-typedef struct _zval_struct zval;
+/* Intentionally avoid typedef because of C99 redeclaration errors. */
+struct _zend_array;
```

Comment:
Seeing this, I already wondered before: Why is the struct name even prefixed with an underscore instead of matching the typedef name exactly?

Reply:
I think this is just an convention to avoid having identical names. And maybe also as a hint that clients should use the typedef'd name.